client/document/desktop.jsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import React, { Fragment } from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { jsonStringifyForHtml } from '../../server/sanitize';
```

Comment:
completely unrelated to this PR, so definitely don't take any of this as an attack against you @flootr but heres a totally undeserved rant :) 

---
What is this function! Based on the directory `sanitize` and the function name it seems like its for sanitizing JSON for security purposes. 


So my first question: do we need to sanitize the json?
The app json is coming from `server/pages/index.js` and the code looks like:

```jsx
	context.app = {
		// use ipv4 address when is ipv4 mapped address
		clientIp: request.ip ? request.ip.replace( '::ffff:', '' ) : request.ip,
		isDebug: context.env === 'development' || context.isDebug,
		staticUrls: staticFilesUrls,
	};
```


I'm no good at security questions so I genuinely don't know the answer to the question, but lets say we _do_ need to sanitize it the json.  The function `jsonStringifyForHtml` doesn't have any tests so how do we even know its working!

</EndRant>


---

Split out issue : https://github.com/Automattic/wp-calypso/issues/22300

Reply:
thanks for caring ❤️ and extracting an issue for that 👍